src/utils.py

```diff
@@ -59,3 +69,63 @@ def get_repo_approvers():
         data = yaml.load(fd.read(), Loader=yaml.SafeLoader)
 
     return data["approvers"]
+
+
+def set_commit_status_success_semver(commit):
+    commit.create_status(
+        state="success",
+        description="Semver label exists",
+        context=BLOCK_MERGE_SEMVER_CONTEXT,
+    )
+
+
+def set_commit_status_pending_no_semver(commit):
+    commit.create_status(
+        state="pending",
+        description="Missing Semver label",
+        context=BLOCK_MERGE_SEMVER_CONTEXT,
+    )
+
+
+def check_body_for_semver_labels(body):
+    new_labels = []
+    remove_labels = []
+    if re.match(r"^.*\B/major\b.*$", body):
+        new_labels = LABELS_SEMVER_MAJOR
+        remove_labels = LABELS_SEMVER_MINOR + LABELS_SEMVER_PATCH
+    elif re.match(r"^.*\B/minor\b.*$", body):
+        new_labels = LABELS_SEMVER_MINOR
+        remove_labels = LABELS_SEMVER_MAJOR + LABELS_SEMVER_PATCH
+    elif re.match(r"^.*\B/patch\b.*$", body):
+        new_labels = LABELS_SEMVER_PATCH
+        remove_labels = LABELS_SEMVER_MAJOR + LABELS_SEMVER_MINOR
+
+    return new_labels, remove_labels
+
+
+def semver_labels_exist_in_pr(pull):
+    current_labels = get_labels(pull=pull)
+    for label in current_labels:
+        if label.startswith(f"{LABEL_SEMVER_PREFIX}/"):
+            return True
+
+
+def add_remove_labels(pull, new_labels, remove_labels):
```

Comment:
this apply to all labels or only to semver ones?

Reply:
No reason it couldn't be used for others, tried to keep it generic